components/server/src/workspace/workspace-starter.ts

```diff
@@ -486,20 +488,27 @@ export class WorkspaceStarter {
             if (WorkspaceImageSourceDocker.is(imgsrc)) {
                 let source: WorkspaceInitializer;
                 const disp = new DisposableCollection();
-                let checkoutLocation = this.getCheckoutLocation(workspace);
+                let checkoutLocation = CommitContext.is(workspace.context) && workspace.context.checkoutLocation || '.';
```

Comment:
Why is it ok to drop `workspace.config.checkoutLocation` here? :thinking:

Reply:
The checkoutLocation from the config is respected when the context is created.